transaction/transaction.go

```diff
@@ -490,7 +685,148 @@ func MakeAssetConfigTxnWithFlatFee(account string, fee, firstRound, lastRound ui
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetAcceptanceTransaction(account, fee, firstRound, lastRound, note, genesisID, genesisHash, creator, index)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	tx.Fee = types.MicroAlgos(fee)
+
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
+	return tx, nil
+}
+
+// MakeAssetRevocationTransactionWithFlatFee creates a tx for revoking an asset from an account and sending it to another
+// - account is a checksummed, human-readable address; it must be the revocation manager / clawback address from the asset's parameters
+// - target is a checksummed, human-readable address; it is the account whose assets will be revoked
+// - recipient is a checksummed, human-readable address; it will receive the revoked assets
+// - fee is a flat fee
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetRevocationTransactionWithFlatFee(account, target, recipient string, amount, fee, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetRevocationTransaction(account, target, recipient, amount, fee, firstRound, lastRound,
+		note, genesisID, genesisHash, creator, index)
+
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	tx.Fee = types.MicroAlgos(fee)
+
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
+	return tx, nil
+}
+
+// MakeAssetMintTransactionWithFlatFee creates a tx for minting more of an asset (transferring from the reserve account)
+// Note that this is just an asset transfer transaction where sender=reserve
+// - reserve is a checksummed, human-readable address that will send the transaction. it must be the asset's reserve to be considered a minting, but this is not enforced by the function.
+// - recipient is a checksummed, human-readable address; it will receive the revoked assets
+// - amount is the number of assets to send
+// - fee is the flat fee
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetMintTransactionWithFlatFee(reserve, recipient string, amount, fee, firstRound, lastRound uint64, note []byte,
```

Comment:
Maybe have this wrap around MakeAssetTransferTxnWithFlatFee instead? Then you don't need to check for flat fee since MakeAssetTransferTxnWithFlatFee takes care of it.